File: patterns/content-based-router/java-camel/MainTest.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class MainTest {
    @Test
    public void routesOrders() {
        Main.Message msg = new Main.Message("orders", "payload");
        assertEquals("orders", Main.route(msg));
    }

    @Test
    public void routesPayments() {
        Main.Message msg = new Main.Message("payments", "payload");
        assertEquals("payments", Main.route(msg));
    }

    @Test
    public void routesUnknown() {
        Main.Message msg = new Main.Message("other", "payload");
        assertEquals("unknown", Main.route(msg));
    }
}

File: patterns/content-based-router/kotlin-spring-boot/MainTest.kt
import kotlin.test.Test
import kotlin.test.assertEquals

class MainTest {
    @Test
    fun routesOrders() {
        val msg = Message("orders", "payload")
        assertEquals("orders", route(msg))
    }

    @Test
    fun routesPayments() {
        val msg = Message("payments", "payload")
        assertEquals("payments", route(msg))
    }

    @Test
    fun routesUnknown() {
        val msg = Message("other", "payload")
        assertEquals("unknown", route(msg))
    }
}
